lib/puppet_x/puppetlabs/powershell/powershell_manager.rb

```diff
@@ -104,10 +108,21 @@ def make_ps_init_code(init_ready_event_name)
         template.result(binding)
       end
 
-      def make_ps_code(powershell_code, output_ready_event_name, timeout_ms = 300 * 1000)
+      def make_ps_code(powershell_code, output_ready_event_name, timeout_ms = default_execute_timeout, working_dir = nil)
+        # Convert the working directory to a native pathname, if required
+        if (working_dir == nil)
+          set_ps_location = ''
+        else
+          if Puppet::Util::Platform.windows?
+            working_dir = working_dir.gsub(File::SEPARATOR, File::ALT_SEPARATOR)
```

Comment:
Forward and backslashes are not valid characters in directory names so it is safe to convert.  This step was to ensure that we give PowerShell windows native looking paths instead of the Puppet style forward slash based ones.